homeassistant/components/vodafone_station/api.py

```diff
@@ -0,0 +1,396 @@
+            allow_redirects=False,
+        )
+
+        j = await reply.json(content_type="text/html")
+        user_obj = {}
+        for item in j:
+            key = list(item.keys())[0]
+            val = list(item.values())[0]
+            user_obj[key] = val
+
+        self.encryption_key = user_obj["encryption_key"]
+        _LOGGER.debug("encryption_key: <%s>", self.encryption_key)
+
+    async def _encrypt_string(self, credential: str) -> str:
+        """Encrypt username or password for login."""
+
+        credential = urllib.parse.quote(credential)
+        credential = html.unescape(credential)
+        hash1_str = hmac.new(
+            bytes("$1$SERCOMM$", "latin-1"),
+            msg=bytes(credential, "latin-1"),
+            digestmod=hashlib.sha256,
+        ).hexdigest()
+
+        return hmac.new(
+            bytes(self.encryption_key, "latin-1"),
+            msg=bytes(hash1_str, "latin-1"),
+            digestmod=hashlib.sha256,
+        ).hexdigest()
+
+    async def _set_cookie(self) -> None:
+        """Enable required session cookie."""
+        self.session.cookie_jar.update_cookies(
+            SimpleCookie(f"domain={self.host}; name=login_uid; value=1;")
+        )
+
+    async def _reset(self) -> bool:
+        """Reset page content before loading."""
+
+        payload = {"chk_sys_busy": ""}
+        timestamp = datetime.now().strftime("%s")
+        url = f"{self.base_url}/data/reset.json?_={timestamp}&csrf_token={self.csrf_token}"
+        reply = await self.session.post(
+            url,
+            data=payload,
+            headers=self.headers,
+            timeout=10,
+            verify_ssl=False,
+            allow_redirects=False,
+        )
+
+        return reply.status == 200
+
+    async def _overview(self) -> dict:
+        """Load overview page information."""
+        _LOGGER.debug("Getting overview for host %s", self.host)
+        timestamp = datetime.now().strftime("%s")
+        url = f"{self.base_url}/data/overview.json?_={timestamp}&csrf_token={self.csrf_token}"
+
+        reply = await self.session.get(
+            url,
+            headers=self.headers,
+            timeout=10,
+            verify_ssl=False,
+            allow_redirects=True,
+        )
+        reply_json = await reply.json(content_type="text/html")
+        _LOGGER.debug("Full Response (overview): %s", reply_json)
+        return reply_json
+
+    async def _user_data(self) -> dict:
+        """Load user_data page information."""
+        _LOGGER.debug("Getting user_data for host %s", self.host)
+        timestamp = datetime.now().strftime("%s")
+        url = f"{self.base_url}/data/user_data.json?_={timestamp}&csrf_token={self.csrf_token}"
+
+        reply = await self.session.get(
+            url,
+            headers=self.headers,
+            timeout=10,
+            verify_ssl=False,
+            allow_redirects=False,
+        )
+        reply_json = await reply.json(content_type="text/html")
```

Comment:
This content type doesn't look right for json

Reply:
Same as above